### DISCUSSION TITLE
Error with Custom Scalar

### DISCUSSION THREAD

**Discussion by jmbarroso** [Netflix/dgs-framework#198] [Q&A] [answered]
Hi, 

I was trying to test a custom scalar. I followed the [documentation](https://netflix.github.io/dgs/scalars/)  

The problem is that when adding the custom Scalar to a type, the generated Kotlin class shows an "unresolved reference," and I don't see any Kotlin class generated for the new custom scalar.  

My implementation for the scalar included:
- Adding the scalar in the schema 
```
scalar Amount
```
- Coercing Interface implementation 
```
@DgsScalar(name = "Amount")
class AmountScalar : Coercing<BigDecimal,String> {
...
}
```
- Using the new scalar. When using the scalar, the generated Kotlin class does not compile. 
```
type Compensation
{
    id: ID!
    name: String!
    amount: Amount
}
```

I looked into the Kotlin example project, but the scalar added in that project was an extension of the Graphql deprecated types. I suspect that something is missing in my implementation/docs. 

Thank you

**Comment by paulbakker** [comment, marked as answer]
If you have existing Java/Kotlin types you want to use (instead of generating them), you need to have a `typeMapping` in your codegen configuration. 
For example: 

```
generateJava{
   typeMapping = ["MyGraphQLType": "com.mypackage.MyJavaType"]
}
```

That should fix it!

**Reply by jmbarroso** [reply]
Shouldn't a class for the new scalar be generated?. I'm sorry, I'm not sure if I'm not following your answer or if my question was not clear. I tried to add more info to the above question. 

In this case, I want an amount scalar that internally will be a BigDecimal.  

**Edit:** it seems like I need to map my new scalar to an existing class. Adding to the _generateJava_ task something like this:
```
Tasks.withType<com.netflix.graphql.dgs.codegen.gradle.GenerateJavaTask> {
         ...
	typeMapping = mutableMapOf("Amount" to "java.math.BigDecimal")
}
```
For some reason, I was expecting the code-gen to generate an Amount class or infer it was a BigDecimal based on the scalar Input Output definition, but it is not the case.

**Reply by paulbakker** [reply]
Correct, we can't generate scalars. In this case, the scalar is pretty trivial, but that's usually not the case, and some sort of conversion code needs to be implemented.

**Reply by jmbarroso** [reply]
Thanks!
